### PR TITLE
MEN-3677: Add the style-guide to the Mender-docs repo

### PR DESCRIPTION
**Pull request by oleorhagen**
This adds the style-guide from:
https://docs.google.com/document/d/1LbrWKkVXf9s5Lq2Sa74WDb8Okh7h27Ptbf3a16fz57Y/edit
to the Mender-docs repo in markdown.

Changelog: None

Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

File: Styleguide.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Mender Documentation writing style guide
+
+## Why do we need one?
+
+To help us write documentation in a more clear way and keep a consistent tone,
+voice, and style. We have many writers and different reviewers; this guide can
```
Comment (mirzak) [inline Styleguide.md:6]: Can you break this up,

```
To help us write documentation in a more clear way and keep a consistent tone,
voice, and style.

We have many writers and different reviewers...
```

Comment (mirzak) [inline Styleguide.md:6]: `reviewers;` -> `reviewers,`

File: Styleguide.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+be a reference for all to follow. Our audience members have diverse backgrounds
+in terms of language and skills, however we can assume a certain base level of
+reading, comprehension and technical knowledge. Keeping this in mind, we should
+make it as easy as possible for all of our target audience to use our product.
+Consistency and simplicity helps all our readers.
+
+# What does it cover?
```
Comment (mirzak) [inline Styleguide.md:13]: `#` -> `##`

File: Styleguide.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+This documentation style guide borrows from the style guides of IBM
+developerWorks, and Red Hat style guides. It covers the general guidelines of
```
Comment (mirzak) [inline Styleguide.md:16]: `IBM developerWorks, and Red Hat style guides`, these should be links

File: Styleguide.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+Spelling can also be checked with the `spell-check.sh` script.
+
+## Use a single space before a new sentence after periods - not double spaces
+
+Example:
+
+> "First sentence. Second sentence."
+
+## Headers:
```
Comment (mirzak) [inline Styleguide.md:37]: `## Headers:` -> `## Headers`

File: Styleguide.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+Unless one of the following applies:
+
+* The system performs the action.
+* It is more appropriate to focus on the receiver of the action.
+* You want to avoid blaming the user for an error, such as in an error message.
+* The information is clearer in passive voice.
+* Avoid -ing endings if possible as they make the sentence longer and more passive / less actionable. Examples:.
+* "Get started", not "Getting started".
+* "Provision a new device", not "Provisioning a new device".
+* Helpful resource: the section titled "Passive Voice Should Never Be Used by You" in the Linux Journal Author's Guide.
```
Comment (mirzak) [inline Styleguide.md:72]: `"Passive Voice Should Never Be Used by You" in the Linux Journal Author's Guide.` should be a link

File: Styleguide.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+# Useful tools
+
+Punctuation and grammar tools: these apps can help evaluate your writing and warn about common problems such as passive voice, complexity, sentence length etc.
+
+* https://grammark.org/dist/#/
+* http://www.hemingwayapp.com/
+* https://grammarly.com (payware)
+* ./scripts/passive.sh <file>
+* ./scripts/weasel.sh <file>
```
Comment (mirzak) [inline Styleguide.md:161]: Would drop this, until the other PRs are merged and would link to docs describing the tools here.

File: Styleguide.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+# Taxonomy
+
+Our taxonomy guide can be found
+[here](https://docs.google.com/document/d/1Lk6lNxaz8v2ug6OaJVrtAn3j8I7_2ebd9Z0CI9-58IA/edit).
```
Comment (mirzak) [inline Styleguide.md:166]: This links to a private document. Maybe just copy the "Product taxonomy" here?

We also plan to have this as part of docs.mender.io, as is preparerad [here](https://github.com/mendersoftware/mender-docs/blob/documentation-restructuring/02.Overview/12.Taxonomy/docs.md). So maybe we should just wait with this section until it is merged in to docs.mender.io, then we can simply link to it.

File: Styleguide.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+It covers general industry terms as well as product and brand specific words,
+spelling and capitalization.
+
+
+# Miscellaneous 
+
+* External links should always open in a new tab/window
```
Comment (mirzak) [inline Styleguide.md:173]: Would add an examples on how to make a link open in new tab/window

File: Styleguide.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+Example of items:
+> * U-Boot
+> * GRUB
+> * Refer to How to Capitalize and Punctuate Bullet Points for details
+
+# Useful tools
+
+Punctuation and grammar tools: these apps can help evaluate your writing and warn about common problems such as passive voice, complexity, sentence length etc.
+
+* https://grammark.org/dist/#/
```
Comment (mirzak) [inline Styleguide.md:157]: Remove `#/`

File: Styleguide.md (hunk 6 above)
Comment (oleorhagen) [reply Styleguide.md:161]: Which PRs are you referring to, explicitly?

File: Styleguide.md (hunk 7 above)
Comment (oleorhagen) [reply Styleguide.md:166]: Alright, holding off on this one.

File: Styleguide.md (hunk 6 above)
Comment (mirzak) [reply Styleguide.md:161]: https://github.com/mendersoftware/mender-docs/pull/1016

**Comment by mirzak** [comment]
Did you forget about this one @oleorhagen ? :)

**Comment by oleorhagen** [comment]
@mirzak I thought we were supposed to wait until Taxonomy was merged?

**Comment by mirzak** [comment]
You can fix up everything else. We can also followup with the taxonomy link if that is the only thing that remains.

File: README-styleguide.markdown [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+# Useful tools
+
+Punctuation and grammar tools: these apps can help evaluate your writing and warn about common problems such as passive voice, complexity, sentence length etc.
+
+* https://grammark.org
+* http://www.hemingwayapp.com
+* https://grammarly.com (payware)
+* ./scripts/passive.sh <file>
+* ./scripts/weasel.sh <file>
```
Comment (mirzak) [inline README-styleguide.markdown:166]: You can go ahead and provide links for these.

File: README-styleguide.markdown [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+
+# Taxonomy
+
+Our taxonomy reference can be found [here](02.Overview/15.Taxonomy/docs.md). It
+covers general industry terms as well as product and brand specific words,
+spelling and capitalization.
```
Comment (mirzak) [inline README-styleguide.markdown:172]: Taxonomy has been merged.

Would drop:

> It covers general industry terms as well as product and brand specific words, spelling and capitalization.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+[here](./README-bash-styleguide.markdown).
+
+### Tools for documentation writers
+
+The project has a few tools to help with developing documentation:
+
+* [checklinks](README-checklinks.markdown) - Which automatically checks all internal links for duds.
+* [./scripts/passive.sh](./scripts/README-passive.markdown) - Checks a document for passive voice.
+* [./scripts/weasel.sh](./scripts/README-weasel.markdown) - Checks a document for weasel words.
+
```
Comment (mirzak) [inline CONTRIBUTING.md:93]: Would add a reference to https://github.com/mendersoftware/mender-docs/blob/master/README-autoversion.markdown

Comment (mirzak) [reply CONTRIBUTING.md]: This was resolved but I do not see the change?

Comment (oleorhagen) [reply CONTRIBUTING.md]: lol, sry. Added it to the old `Contributing.md` file locally

**Pull request merged**